src/components/ui/autocomplete.tsx

```diff
@@ -45,23 +49,72 @@ export default function Autocomplete({
   onChange,
   onSearch,
   placeholder = "Select...",
+  inputPlaceholder = "Search option...",
   noOptionsMessage = "No options found",
   disabled,
+  icon,
   align = "center",
+  className,
   popoverClassName,
+  freeInput,
   "data-cy": dataCy,
 }: AutocompleteProps) {
   const [open, setOpen] = React.useState(false);
-
   const isMobile = useBreakpoints({ default: true, sm: false });
 
+  // Maintain an internal state for the input text when freeInput is enabled.
+  const [inputValue, setInputValue] = React.useState(value);
```

Comment:
Not a huge fan of this; in most places we'll anyways be keeping another state in the parent component for the search query; 

This change will now unnecessarily maintain two states for the search query on all other usage where free-text is not required; one here and one in the parent component;

This could be simplified by leveraging `mergeAutocompleteOptions` utility along with the currently entered value;